Add core cython hello

jcfr opened the pull request:
Once finalized, this pull request will add the `core-cython-hello` sample project demonstrating how to leverage `scikit-build-core` and `cython`.

In the meantime, it is used as a "maturing" ground to consolidate the changes originally developed by @vyasr in the context of https://github.com/rapidsai/rapids-cmake/pull/433


Related issues:
* https://github.com/scikit-build/scikit-build-core/issues/442
* https://github.com/cython/cython/issues/5486
* https://gitlab.kitware.com/cmake/cmake/-/issues/24982
* https://github.com/rapidsai/rapids-cmake/pull/433

vyasr commented:
@jcfr you can start using https://github.com/vyasr/cython-cmake/ in this example. The code there is ready for some initial stress testing.

jcfr commented:
The failures are associated with the `pi-fortran` sample project and are unrelated to `core-cython-hello` project.

jcfr committed:
feat(core-cython-hello): add sample project using cython-cmake

Add a minimal scikit-build-core + Cython sample that transpiles a .pyx
via the cython-cmake helpers (find_package(Cython) + UseCython +
cython_transpile), and register it in the noxfile test/dist lists.

Original work by @jcfr and @vyasr; rebased onto main and updated to the
current cython-cmake API.

Assisted-by: ClaudeCode:claude-opus-4.8
